text/csv-indexation.md

```diff
@@ -0,0 +1,127 @@
+- Title: Csv Indexation
+- Start Date: 2021-04-9
+- Specification PR: TODO
+- Meilisearch Issue: [#128](https://github.com/meilisearch/transplant/issues/128),[#1332](https://github.com/meilisearch/MeiliSearch/issues/1332)
+
+## Feature Description and Interaction
+
+### Summary
+
+The initiation step of document indexing is to send some file matching a format to be parsed and tokenized in order to give search results to end-users. A [CSV](https://en.wikipedia.org/wiki/Comma-separated_values) file format is broadly used to store and exchange data in a simple format.
+
+Also, in order to boost write performance csv files are more suited than JSON for huge datasets.
+
+### Motivation
+
+We want to provide our users with an always improved usage experience. Currently, the engine only accepts JSON format as a data source. We want to give users the possibility of another simple file format, well known, to use. Thus, give them more versatility at the data source choices for the indexation (add and update) step.
+
+Since most of SQL engines or SQL clients can easily dump data as a csv file, it will facilitate MeiliSearch adoption by extending the indexation step on a wider range of customer cases than before.
+
+Writing performance is also considered as a motivation since CSV parsing is less CPU and memory intensive than parsing Json.
```

Comment:
The main issue with json is that you need to hold the entirety on the deserialized payload in memory before you can start ingesting. This is a huge overhead.

Reply:
Thanks, I will update the specs to mention that in this part.